packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx

```diff
@@ -114,6 +115,53 @@ export class ChromeService {
     );
   }
 
+  private setIsVisible = (isVisible: boolean) => this.isForceHidden$.next(!isVisible);
+
+  /**
+   * Some EUI component can be toggled in Full screen (e.g. the EuiDataGrid). When they are toggled in full
+   * screen we want to hide the chrome, and when they are toggled back to normal we want to show the chrome.
+   */
+  private handleEuiFullScreenChanges = () => {
+    const { body } = document;
+    // HTML class names that are added to the body when Eui components are toggled in full screen
+    const classesOnBodyWhenEuiFullScreen = ['euiDataGrid__restrictBody'];
```

Comment:
Discussion idea: should we work with EUI team to find a "contractual" className or attribute added to the body tag when EUI components go full screen? I'm asking, because this className has the look of something internal to EUI.

Reply:
Yes that is a good idea 👍 

As you can see, the security team already hardcodes this class name ([here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/common/containers/use_full_screen/index.tsx#L35), [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/timelines/components/timeline/session_tab_content/use_session_view.tsx#L143)) and the discover team uses it to extend some CSS ([here](https://github.com/elastic/kibana/blob/main/src/plugins/discover/public/embeddable/saved_search_grid.scss#L14))

@elastic/eui-team  What do you think about defining a common class that is added to the `body` whenever any EUI component enters full screen mode? And then exposing it as a constant?

Reply:
Great idea! However, to throw a weird wrench in things, I think EuiDataGrid _may_ be the only component that has this behavior in its full screen mode. All other components that have a full screen mode (modals/flyouts, EuiCodeBlock, and EuiImage) do not need this affordance as far as I can tell. Am I off on that? I haven't peeked at the full code changes so I'm not sure if I'm missing some extra context here.

Reply:
Ok so we could hold on before doing it. And the moment we see that the `classesOnBodyWhenEuiFullScreen ` array needs another class for a different component we can revisit the decision?

Reply:
Sounds great. Thanks for the flexibility!